### ISSUE TITLE
Inconsistency between MoneyField and CurrencyField

### ISSUE THREAD

**Issue by washeck** [django-money/django-money#487]
Hi,

it seems like there is an inconsistency (or at least a confusing behavior) in how currency is represented by `CurrencyField` and `MoneyField` (or better say `Money` object).

Let's have a simple model

```
class Account(models.Model):
    currency = CurrencyField()
    balance = MoneyField(max_digits=20, decimal_places=4, null=True)

account = Account(currency='EUR', balance=Money(20, 'EUR'))
account.save()
```

Now if I compare balance currency and the currency of the account, it is surprisingly not equal

```
>>> account.balance.currency == account.currency 
False
```

The reason is obviously differrency in types
```
>>> type(account.currency)
str
>> type(account.balance.currency)
moneyed.classes.Currency
```
and the way how equality is implemented in `moneyed.Currency`
```
    def __eq__(self, other):
        return type(self) is type(other) and self.code == other.code
```

I know I can make the comparison using currency code
```
>>> account.balance.currency.code == account.currency 
True
```
but I find it confusing and very error prone - if you forget to add the `code` attribute you get no error but the comparison is just False.

I would expect the `CurrencyField` to return instance of `Currency` instead of string. I know this is highly backwards incompatible but to be honest otherwise I don't see much reason for existence of `CurrencyField` which is just shortcut for `CharField(max_lenght=3)`.

Another option would be at least to make `moneyed.Currency` raise `TypeError` on comparison to str but I'm not sure if this has a chance to be merged to moneyed.

**Comment by washeck** [comment]
Btw, looking at the fact that `CurrencyField` is not included in `djmoney.models.fields.__all__` I'd like to ask if it is meant to be used by user code or is it just an implementation detail of `MoneyField`.

**Comment by Stranger6667** [comment]
Hej @washeck !

Good points! 
Yep, I agree. It would be better to return a `Currency` instance from `CurrencyField`, it is more in line with the `moneyed` behavior. Would be nice to have it as a part of 1.0 release.

About `CurrencyField` - it is not intended to be used explicitly :) so, an implementation detail

@benjaoming 
Ben, what do you think?

**Comment by benjaoming** [comment]
I agree that the rendered type for `CurrencyField` should be a `Currency` object. The internal db representation can be a string.

**Comment by benjaoming** [comment]
@Stranger6667 I am moving this til a 1.1 release so 1.0 isn't stuck any more. This is the only out-standing issue.

**Comment by Stranger6667** [comment]
Sure, Ben :) thank you. At the moment I don’t have much capacity for django-money, but hope to have it in a few weeks, so I could be more involved

**Comment by piotrsynowiec** [comment]
Hi Guys,

I stumbled upon a similar problem. 

Wanted to create a model for Currencies based on CurrencyField (to have a list of Currencies for a Company, that sets the BankAccounts currencies list later on), and that went just fine, but then when I wanted to create a serializer, I noticed that there's no CurrencyField in  djmoney.contrib.django_rest_framework and I can't show this field in REST API.

Is there any workaround for that? Or the only way is to restructure the models and resign from external Currency model?

Thanks so much for your help.
